File: Binary Tree Postorder Traversal.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return a list of integers
    def postorderTraversal(self, root):
        ret = []
        if (root == None):
            return ret
        if (root.left != None):
            left = self.postorderTraversal(root.left)
            ret.extend(left)
        if (root.right != None):
            right = self.postorderTraversal(root.right)
            ret.extend(right)
        ret.append(root.val)
        return ret
        

File: Binary Tree Postorder Traversal.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode *root) {
        vector<int> ret;
        if (!root)
            return ret;
        if (root->left) {
            vector<int> left = postorderTraversal(root->left);
            ret.insert(ret.end(), left.begin(), left.end());
        }
        if (root->right) {
            vector<int> right = postorderTraversal(root->right);
            ret.insert(ret.end(), right.begin(), right.end());
        }
        ret.push_back(root->val);
        return ret;
    }
};